### PR TITLE
src/Objecter.cc: Don't be redundant with == or != in conditions.

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -1544,7 +1544,7 @@ void Objecter::_check_op_pool_dne(Op *op, unique_lock *sl)
 
       OSDSession *s = op->session;
       if (s) {
-	assert(s != NULL);
+	assert(s);
```
Comment (joscollin) [inline src/osdc/Objecter.cc]: What is the use of an assert here, as `s` always evaluates to `true` ?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-11-18_15:01:48-rados-wip-kefu-testing-2017-11-18-2124-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-20_05:45:10-rados-wip-kefu-testing-2017-11-18-2124-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-20_06:28:09-rados-wip-kefu-testing-2017-11-18-2124-distro-basic-mira/

File: src/osdc/Objecter.cc (hunk 1 above)
Comment (tchaikov) [reply src/osdc/Objecter.cc]: @shinobu-x could you take this chance to remove this `assert()`?

**Comment by tchaikov** [comment]
closing as lack of activity.

**Pull request closed without merging**